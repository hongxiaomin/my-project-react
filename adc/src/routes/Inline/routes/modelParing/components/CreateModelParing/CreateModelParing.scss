  .dialog {
    width:500px;
    padding:2px;
    background: #F7F7F7;
  }

  .header {
    height: 65px;
  }

  .onlyHeaderTitle {
    margin: 25px 0px 10px 15px;
  }

  .headerTitle {
    margin: 10px 0px 10px 15px;
    font-size: 16px;
  }

  .headerID {
    margin-left: 15px;
    font-size: 18px;
  }
  .dialogBody {
    background: #fff;
    color: #000;
  }
  .filter {
    border-bottom: solid 1px #C9C8C9;
    display: flex;
    align-items: center;
    justify-content: space-between;
    height:65px;
    line-height:60px;
    margin:0px 10px;
    font-size: 14px;
  }
  .filterDefect {
    border-bottom: solid 1px #C9C8C9;
    display: flex;
    align-items: center;
    justify-content: space-between;
    // height:65px;
    // line-height:60px;
    margin:0px 10px;
    font-size: 14px;
  }
  .filter input {
    width: 200px;
  }
  .select {
    cursor: pointer;
    outline: 0;
    border: 0;
    background: inherit;
    box-shadow: none;
    margin: 0;
    border: 1px solid #D9D9D9;
    height: 30px;
    width: 200px;
    border-radius: 3px;
  }
  .btnGroup {
    float: right;
    margin:17px 0px;
  }
  .btnEdit {
    background: #0087DC;
    color: #fff;
    margin: 0px 10px 0px 15px;
    width: 80px;
    height: 35px;
    font-size: 14px;
  }
  .btnCancel {
    background: #fff;
    border: 1px solid #0087DC !important;
    color: #0087DC !important;
    width: 80px;
    height: 35px;
    font-size: 14px;
  }
