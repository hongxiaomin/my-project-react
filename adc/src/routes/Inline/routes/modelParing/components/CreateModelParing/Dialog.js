import React from 'react'
import PropTypes from 'prop-types'
import WrapperFormattedMessage from 'components/WrapperFormattedMessage'
import { injectIntl } from 'react-intl'
import { isEmpty } from 'lodash'
import { Button, Dialog } from '@blueprintjs/core'
import { Select, Checkbox } from 'antd'
import classes from './CreateModelParing.scss'
import { getModelOption, CHANGE_STATUS, MODEL_STATUS } from '../../modules/ModelParing'

const Option = Select.Option
export class DialogFilter extends React.Component {

  constructor(props) {
    super(props)
    this.state = {

    }
    this.handleChange = this.handleChange.bind(this)
    this.trim = this.trim.bind(this)
    this.onBlur = this.onBlur.bind(this)
    this.checkboxDefault = this.checkboxDefault.bind(this)
  }

  trim(str) {
    if (!str) return ''
    return str.replace(/(^\s*)|(\s*$)/g, '')
  }

  onBlur() {
    if (!isEmpty(this.props.that.state.processId) &&
     !isEmpty(this.props.that.state.productId)) {
      this.props.dispatch(getModelOption(this.props.that.state.processId,
         this.props.that.state.productId))
      this.props.dispatch({
        type: CHANGE_STATUS,
        payload: false,
      })
    }
    if (!isEmpty(this.props.that.state.processId) &&
     isEmpty(this.props.that.state.productId)) {
      this.props.dispatch(getModelOption(this.props.that.state.processId, ''))
      this.props.dispatch({
        type: CHANGE_STATUS,
        payload: false,
      })
    }
  }
  // 是否设置为试运行
  checkboxDefault(e) {
    this.props.that.setState({
      isTrialRun: e.target.checked,
    })
  }
  handleChange(type) {
    return (value) => {
      if (type === 'model') {
        this.props.dispatch({
          type: MODEL_STATUS,
          payload: value,
        })
      }
      this.props.that.setState({
        [type]: this.trim(value),
      })
    }
  }

  render() {
    const { intl } = this.props
    let defectCode
    if (!isEmpty(this.props.modelstatus) && !isEmpty(this.props.modelOption.modelList)) {
      defectCode = this.props.modelOption.modelList.filter((item) => {
        return item.modelId.toString() === this.props.that.state.model
      })
      defectCode = defectCode[0].defectList.join('/')
    }
    if (isEmpty(this.props.modelOption.modelList) && isEmpty(this.props.modelstatus)) {
      defectCode = []
    }
    return (
      <div>
        <Dialog
          isOpen={this.props.isOpen}
          onClose={this.props.toggleDialog}
          className={classes.dialog}
        >
          <div className={classes.header}>
            <h6 className={this.props.DialogID === '' ? classes.onlyHeaderTitle : classes.headerTitle}>
              {this.props.DialogTitle}
            </h6>
          </div>
          <div className={classes.dialogBody}>
            <div className={classes.filter}>
              <WrapperFormattedMessage id='processId' />
              <Select
                allowClear
                showSearch
                name='processId'
                optionFilterProp="children"
                style={{ width: 200 }}
                placeholder={intl.formatMessage({ id: 'pleaseSelect' })}
                onChange={this.handleChange('processId')}
                onBlur={this.onBlur}
              >
                {this.props.processId.map(item => (<Option
                  key={item.processName}
                  value={item.processId.toString()}
                >{item.processName}</Option>))}
              </Select>
            </div>
            <div className={classes.filter}>
              <WrapperFormattedMessage id='productId' />
              <Select
                allowClear
                showSearch
                optionFilterProp="children"
                name='productId'
                style={{ width: 200 }}
                placeholder={intl.formatMessage({ id: 'pleaseSelect' })}
                onChange={this.handleChange('productId')}
                onBlur={this.onBlur}
              >
                {this.props.productId.map(item => (<Option
                  key={item.productId}
                  value={item.productId.toString()}
                >{item.productName}</Option>))}
              </Select>
            </div>
            {
              !isEmpty(this.props.checkError) &&
              <div style={{ color: 'red', margin: '0 10px' }}>Tips：{this.props.checkError}</div>
            }
            <div className={classes.filter}>
              <WrapperFormattedMessage id='modelName' />
              <Select
                showSearch
                name='model'
                optionFilterProp="children"
                style={{ width: 200 }}
                placeholder={intl.formatMessage({ id: 'pleaseSelect' })}
                onChange={this.handleChange('model')}
              >
                {
                  !isEmpty(this.props.modelOption.modelList) ?
                  this.props.modelOption.modelList.map(item => (<Option
                    key={item.modelId}
                    value={item.modelId.toString()}
                  >{item.modelName}</Option>)) :
                  [].map(item => (<Option
                    key={item.modelId}
                    value={item.modelId.toString()}
                  >{item.modelName}</Option>))
                }
              </Select>
            </div>
            {
              this.props.status === true &&
              !isEmpty(this.props.modelstatus) ?
                <div className={classes.filterDefect}>
                  <WrapperFormattedMessage id='labelingTool.eda.defectCode' />
                  <div style={{ width: '200px',
                    wordWrap: 'break-word',
                    backgroundColor: '#eee',
                    margin: '10px',
                    borderRadius: '5px',
                    padding: '10px' }}
                  >
                    {defectCode}
                  </div>
                </div> :
                <div />
            }
            <div className={classes.filter}>
              <WrapperFormattedMessage id='modelPairing.dialogFilter.trialRun' />
              <Checkbox
                style={{ width: '200px', marginLeft: '50px' }}
                checked={this.props.that.state.isTrialRun}
                onChange={this.checkboxDefault}
              ><WrapperFormattedMessage id='modelPairing.dialogFilter.setAsTrialRun' /></Checkbox>
            </div>
            <div className={classes.btnGroup}>
              <Button className={classes.btnCancel} onClick={this.props.toggleDialog}>
                <WrapperFormattedMessage id='cancel' />
              </Button>
              <Button
                className={`pt-button pt-intent-primary ${classes.btnEdit}`}
                onClick={this.props.handleSubmit}
              >
                <WrapperFormattedMessage id='Save' />
              </Button>
            </div>
          </div>
        </Dialog>
      </div>
    )
  }
}

DialogFilter.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  DialogID: PropTypes.string.isRequired,
  DialogTitle: PropTypes.string.isRequired,
  processId: PropTypes.array.isRequired,
  productId: PropTypes.array.isRequired,
  modelOption: PropTypes.object.isRequired,
  modelstatus: PropTypes.string.isRequired,
  status: PropTypes.bool.isRequired,
  toggleDialog: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  that: PropTypes.object.isRequired,
  dispatch: PropTypes.func.isRequired,
  checkError: PropTypes.string.isRequired,
  intl: PropTypes.object.isRequired,
}

export default injectIntl(DialogFilter)
