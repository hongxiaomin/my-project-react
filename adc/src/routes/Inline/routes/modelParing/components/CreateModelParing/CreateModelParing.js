import React from 'react'
import PropTypes from 'prop-types'
import { injectIntl } from 'react-intl'
import { connect } from 'react-redux'
import Dialog from './Dialog'
import { GET_MODEL_OPTION, MODEL_STATUS, getSelectInfo, addProcessProduct } from '../../modules/ModelParing'

export class CreateModelParing extends React.Component {
  constructor(props) {
    super(props)
    this.handleSubmit = this.handleSubmit.bind(this)
    this.state = {
      processId: '',
      productId: '',
      model: this.props.modelstatus,
      isTrialRun: true,
    }
  }
  componentDidMount() {
    this.props.dispatch(getSelectInfo())
  }

  handleSubmit() {
    const tmp = {
      modelId: this.state.model.toString(),
      processId: this.state.processId,
      productId: this.state.productId,
      isTrialRun: this.state.isTrialRun,
    }

    this.props.dispatch(addProcessProduct(tmp))
    this.setState({
      processId: '',
      productId: '',
      model: '',
      isTrialRun: true,
    })
    this.props.toggleDialog()
    this.props.dispatch({
      type: GET_MODEL_OPTION,
      payload: [],
    })
    this.props.dispatch({
      type: MODEL_STATUS,
      payload: '',
    })
  }

  render() {
    return (
      <div>
        <Dialog
          DialogTitle={this.props.intl.formatMessage({ id: 'modelPairing.createModelParing.createTitle' })}
          DialogID=''
          DialogBtnContent='Create'
          processId={this.props.processList}
          productId={this.props.productList}
          isOpen={this.props.isOpen}
          toggleDialog={this.props.toggleDialog}
          handleSubmit={this.handleSubmit}
          that={this}
          modelOption={this.props.modelOption}
          dispatch={this.props.dispatch}
          status={this.props.status}
          checkError={this.props.checkError}
          modelstatus={this.props.modelstatus}
        />
      </div>
    )
  }
}

CreateModelParing.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  toggleDialog: PropTypes.func.isRequired,
  dispatch: PropTypes.func.isRequired,
  modelOption: PropTypes.object.isRequired,
  status: PropTypes.bool.isRequired,
  modelstatus: PropTypes.string.isRequired,
  productList: PropTypes.array.isRequired,
  processList: PropTypes.array.isRequired,
  checkError: PropTypes.string.isRequired,
  intl: PropTypes.object.isRequired,
}

export default connect(
  ({ auth, modelParing, intl: { locale } }) => ({
    locale,
    token: auth.token,
    modelOption: modelParing.modelOption,
    status: modelParing.status,
    productList: modelParing.productList,
    processList: modelParing.processList,
    modelstatus: modelParing.modelstatus,
    checkError: modelParing.checkError,
  }),
)(injectIntl(CreateModelParing))
