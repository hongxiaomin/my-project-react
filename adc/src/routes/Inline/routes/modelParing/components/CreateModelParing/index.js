import CreateModelParing from './CreateModelParing'

export default CreateModelParing
