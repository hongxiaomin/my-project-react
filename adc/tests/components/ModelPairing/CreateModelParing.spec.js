import React from 'react'
import { CreateModelParing } from 'routes/Inline/routes/ModelParing/components/CreateModelParing/CreateModelParing'

describe('(Component) CreateModelParing', () => {
  const isOpen = false
  const status = false
  const modelstatus = 'a'
  const checkError = 'a'
  const processList = []
  const productList = []
  const modelOption = []
  const dispatch = jest.fn()
  const toggleDialog = jest.fn()
  it('render normally', () => {
    const wrapper = shallow(
      <CreateModelParing
        isOpen={isOpen}
        status={status}
        modelstatus={modelstatus}
        checkError={checkError}
        dispatch={dispatch}
        processList={processList}
        productList={productList}
        modelOption={modelOption}
        toggleDialog={toggleDialog}
      />,
    )

    //  Dialog节点找不到
    // wrapper.find('Dialog').simulate('click')
    // expect(wrapper.state('processId')).toEqual('')
    // expect(wrapper.state('productId')).toEqual('')
    // expect(wrapper.state('model')).toEqual('')
    // expect(wrapper.state('isTrialRun')).toEqual(true)
    expect(wrapper).toMatchSnapshot()
  })
})
