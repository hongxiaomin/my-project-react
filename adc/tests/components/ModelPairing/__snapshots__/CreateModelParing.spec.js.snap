// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) CreateModelParing render normally 1`] = `
<div>
  <DialogFilter
    DialogBtnContent="Create"
    DialogID=""
    DialogTitle="Create Model Pairing"
    checkError="a"
    dispatch={[Function]}
    handleSubmit={[Function]}
    isOpen={false}
    modelOption={Array []}
    modelstatus="a"
    processId={Array []}
    productId={Array []}
    status={false}
    that={
      CreateModelParing {
        "_reactInternalInstance": ShallowComponentWrapper {
          "_calledComponentWillUnmount": false,
          "_compositeType": 0,
          "_context": Object {},
          "_currentElement": <CreateModelParing
            checkError="a"
            dispatch={[Function]}
            isOpen={false}
            modelOption={Array []}
            modelstatus="a"
            processList={Array []}
            productList={Array []}
            status={false}
            toggleDialog={[Function]}
      />,
          "_debugID": 1,
          "_hostContainerInfo": null,
          "_hostParent": null,
          "_instance": [Circular],
          "_mountOrder": 1,
          "_pendingCallbacks": null,
          "_pendingElement": null,
          "_pendingForceUpdate": false,
          "_pendingReplaceState": false,
          "_pendingStateQueue": null,
          "_renderedComponent": NoopInternalComponent {
            "_currentElement": <div>
              <DialogFilter
                      DialogBtnContent="Create"
                      DialogID=""
                      DialogTitle="Create Model Pairing"
                      checkError="a"
                      dispatch={[Function]}
                      handleSubmit={[Function]}
                      isOpen={false}
                      modelOption={Array []}
                      modelstatus="a"
                      processId={Array []}
                      productId={Array []}
                      status={false}
                      that={[Circular]}
                      toggleDialog={[Function]}
              />
      </div>,
            "_debugID": 2,
            "_renderedOutput": <div>
              <DialogFilter
                      DialogBtnContent="Create"
                      DialogID=""
                      DialogTitle="Create Model Pairing"
                      checkError="a"
                      dispatch={[Function]}
                      handleSubmit={[Function]}
                      isOpen={false}
                      modelOption={Array []}
                      modelstatus="a"
                      processId={Array []}
                      productId={Array []}
                      status={false}
                      that={[Circular]}
                      toggleDialog={[Function]}
              />
      </div>,
          },
          "_renderedNodeType": 0,
          "_rootNodeID": 0,
          "_topLevelWrapper": null,
          "_updateBatchNumber": null,
          "_warnedAboutRefsInRender": false,
        },
        "context": Object {},
        "handleSubmit": [Function],
        "props": Object {
          "checkError": "a",
          "dispatch": [Function],
          "isOpen": false,
          "modelOption": Array [],
          "modelstatus": "a",
          "processList": Array [],
          "productList": Array [],
          "status": false,
          "toggleDialog": [Function],
        },
        "refs": Object {},
        "state": Object {
          "isTrialRun": true,
          "model": "a",
          "processId": "",
          "productId": "",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      }
    }
    toggleDialog={[Function]}
  />
</div>
`;
